### PR TITLE
experimental lua template support

### PR DESCRIPTION
**Pull request by contrun**

### CODE REVIEW

File: src/recipe/lua.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    fn bin_name(name: &str) -> String {
+        name.to_string()
+    }
+
+    fn src_template() -> &'static str {
+        "standalone/contract/example.lua"
+    }
+
+    fn build_template() -> &'static str {
+        "standalone/contract/BUILD"
```
Comment (liuck8080) [inline src/recipe/lua.rs:51]: why build is in uppercase

Comment (contrun) [inline src/recipe/lua.rs:51]: It is initially copied that from the c shared library template. I can change that if it is wanted.

**Review by liuck8080 — APPROVED** [review]

File: src/recipe/lua.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    }
+
+    fn create_contract(
+        &self,
+        contract: &Contract,
+        rewrite_config: bool,
+        _signal: &Signal,
+        _docker_env_file: String,
+    ) -> Result<()> {
+        // setup c environment if needed
```
Comment (quake) [inline src/recipe/lua.rs]:
```suggestion
        // setup lua environment if needed
```

File: templates/lua/embedded/contract/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+phf = "0.8"
+
+[build-dependencies]
+includedir_codegen = "0.6"
+blake2b-rs = "0.2"
+faster-hex = "0.6.0"
+
+[dev-dependencies]
+byteorder = "1.3.1"
+ckb-chain-spec = "0.103.0"
```
Comment (quake) [inline templates/lua/embedded/contract/Cargo.toml]: suggest update ckb related crate version to latest 0.107

**Review by quake — APPROVED** [review]

**Pull request merged**